### PR TITLE
feat(i18n): add internationalization support to web-app

### PR DESCRIPTION
**Pull request by lvyueyang**
Hello, I hope to continue to promote the internationalization of `minio`, which is what many people in the community are looking forward to.

I have submitted a PR(https://github.com/minio/console/pull/2443)  before, but it was not passed. I think the main reason for the failure is that there were too many file changes involved, and such large-scale changes would make code review very difficult.

So, this time I want to change it. I will briefly talk about my ideas and hope to use a gradual approach to promote internationalization.

1. Add `i18next` to the project and use only one file for internationalization as an example (control the file changes within 10 files).

2. Add the internationalization extraction tool `i18next-parser`, which can automatically help us maintain the translation directory.

3. Divide the internationalization of the file into many times, and ensure that the file changes are within 10 each time (only some displayed text is changed to `t("text")`).

This will not burden your developers' work. You can only use English for development (only t("text") is needed), and the translation work can be handed over to the community.

If you are worried about the quality of the community's translation, I can withdraw the submission of the `web-app/public/locals/cn/` directory and let the users of `minio` provide their own translations or the community maintain a repository of internationalized translation files. In this way, you only need to add internationalization support in the code and do not need to take any responsibility for the translation results.

Well, this is my idea. If you think this method is good, I hope you can merge this PR after consideration. Everyone is really looking forward to `minio` providing internationalization support. `minio` does a really good job. We also hope to contribute to the beautiful `minio`. Thank you.

### CODE REVIEW

**Comment by Youssef-Harby** [comment]
Any updates on this? Because i want to contribute the Arabic language with RTL support..
Thanks

File: web-app/src/screens/Console/valid-routes.tsx [hunk 1]
```diff
@@ -181,42 +182,42 @@ export const validRoutes = (
...
         {
-          name: "Metrics",
+          name: t("Metrics"),
           id: "monitorMetrics",
           path: IAM_PAGES.DASHBOARD,
           icon: <MetricsMenuIcon />,
         },
         {
-          name: "Logs ",
+          name: t("Logs "),
```
[the remark this answers is not in the document]
Comment (lvyueyang) [reply web-app/src/screens/Console/valid-routes.tsx]: OK, this space appeared 2 years ago, maybe it was a typo? I removed this space in i18n this time, and re-merged the latest master to resolve the conflict.

**Comment by ravindk89** [comment]
We want to start by saying how much we appreciate the community effort here around i18n. We also want to emphasize that we do believe in and support the efforts to improve accessibility across multiple languages.

That said, the baseline for us taking in community contributions is whether or not we can maintain those contributions. i18n is particularly difficult, because we have limited methods for reviewing, approving, and updating content in other languages. 

The maintenance burden ultimately falls on volunteers for a given language, and that doesn't tend to work in the long-run. We've run into this with i18n on our docs products a few years back, and the skew became severe enough that we had to stop support for it entirely. So at least for this approach, we have to decline with the utmost respect for your efforts.

The suggestion for supporting a plugin system is interesting, but we don't have the bandwidth to implement it ourselves - and if a community member steps in, we cannot right now guarantee that we'd have the bandwidth to review/maintain it moving forward.

Thank you again

**Comment by thautwarm** [comment]
Maintenance is a good concern, but maybe adding an interface without an implementation would ease downstream works while keeping the same maintainability.

**Pull request closed without merging**